docs/docs/Tutorials/ROS2/PythonTemplate.md

```diff
@@ -325,6 +325,90 @@ You may also send commands from within the `Controller` class:
 `self.send_pc_scale_command(scale_factor, blocking=False)`  
 `self.send_pc_retract_command(retract_factor, blocking=False)`  
 
+In the `Controller` constructor, you may also uncomment lines 55 or 56 to set the publish rates for
+the Spring or Power Controllers on the buoy. These controllers default to publishing at 10Hz. You
+can call commands to set the rates anywhere from 10Hz to 50Hz (default argument is 50Hz).
+
+``` py linenums="46"
+    def __init__(self):
+        super().__init__('controller')
+
+        self.policy = ControlPolicy()
+        self.set_params()
+
+        # set packet rates from controllers here
+        # controller defaults to publishing @ 10Hz
+        # call these to set rate to 50Hz or provide argument for specific rate
+        # self.set_pc_pack_rate_param()  # set SC publish rate to 50Hz
+        # self.set_sc_pack_rate_param()  # set PC publish rate to 50Hz
+```
+
+## Compile, Test, Run
+
+At this point, your new package should build, pass tests, and run against the sim (will connect
+but do nothing).
+
+From your workspace:
+```
+$ colcon build
+Starting >>> mbari_wec_template_py
+--- stderr: mbari_wec_template_py                   
+/usr/lib/python3/dist-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
+  warnings.warn(
+---
+Finished <<< mbari_wec_template_py [0.74s]
+
+Summary: 1 package finished [0.89s]
+  1 package had stderr output: mbari_wec_template_py
+```
+
+```
+$ colcon test
+Starting >>> mbari_wec_template_py
+--- stderr: mbari_wec_template_py                   
+
+=============================== warnings summary ===============================
+test/test_flake8.py::test_flake8
+test/test_flake8.py::test_flake8
+  Warning: SelectableGroups dict interface is deprecated. Use select.
+
+-- Docs: https://docs.pytest.org/en/stable/warnings.html
+---
+Finished <<< mbari_wec_template_py [0.74s]
+
+Summary: 1 package finished [0.87s]
+  1 package had stderr output: mbari_wec_template_py
+```
+
+Next, run the sim in another terminal:
+`$ ros2 launch buoy_gazebo mbari_wec.launch.py`
+
+Then, in another terminal, source your workspace, and launch the empty controller:
+```
+$ source ~/buoy_ws/install/local_setup.bash
+$ ros2 launch <your_package_name> controller.launch.py
```

Comment:
Question, if they just run this, do they need to turn off any existing controller in buoy_sim that's already running that could be publishing messages to the same topics and clashing with the custom controller?

Reply:
No, the the plugins in buoy_sim are made so that they may be overridden by an external controller. (But you wouldn't want two different _**external**_ controllers commanding the same service...)